lib/client/api.go

```diff
@@ -2308,12 +2309,35 @@ func (tc *TeleportClient) SFTP(ctx context.Context, args []string, port int, opt
 	}
 
 	if !quiet {
-		config.cfg.ProgressWriter = tc.Stdout
+		config.cfg.ProgressWriter = func(fileInfo os.FileInfo) io.Writer {
+			return newProgressBar(fileInfo.Size(), fileInfo.Name(), tc.Stdout)
+		}
 	}
 
 	return trace.Wrap(tc.TransferFiles(ctx, config.hostLogin, config.addr, config.cfg))
 }
 
+// newProgressBar returns a new progress bar that writes to writer.
+func newProgressBar(size int64, desc string, writer io.Writer) *progressbar.ProgressBar {
```

Comment:
I think it might be actually better to put this as an exported function in `lib/sshutils/sftp/sftp.go` and then call it in `lib/client/api.go`. Then `sftp` would provide a "default" implementation of the progress bar that other places can use but callers can still provide their own custom implementation.

`lib/client/api.go` is already quite big.

Reply:
Sounds good, done.